cache/async_cache.go

```diff
@@ -29,22 +32,39 @@ func (c *AsyncCache) Close() error {
 	return nil
 }
 
-func (c *AsyncCache) AwaitForConcurrentTransaction(key *Key) bool {
-	startTime := time.Now()
+type TransactionResult struct {
+	ElapsedTime time.Duration
```

Comment:
I'm a bit picky but since the duration is not used (anymore) outside the function AwaitForConcurrentTransaction, there is no need to keep this field. 
In fact, we could remove the TransactionResult structure and the AwaitForConcurrentTransaction could return just a TransactionState.